app/models/service_order.rb

```diff
@@ -4,12 +4,31 @@ class ServiceOrder < ActiveRecord::Base
   has_many :miq_requests, :dependent => :nullify
 
   validates :state, :inclusion => {:in => %w(wish cart ordered)}
-  validates :name, :presence => true
   validates :state, :presence => true
+  validates :name, :presence => true, :on => :update
 
-  before_create :assign_user_name
+  before_create :assign_user_name, :assign_tenant
+  after_create  :create_order_name
 
   def assign_user_name
     self.user_name = user.try(:name)
   end
+
+  def assign_tenant
+    self.tenant = tenant
+  end
+
+  def create_order_name
+    update_attributes(:name => "Order # #{id}") if name.blank?
+  end
+
+  def checkout
+    _log.info("Service Order checkout for service: #{name}")
+
+    miq_requests.each do |request|
+      request.update_attributes(:process => true)
+      request.call_automate_event_queue("request_created")
```

Comment:
@gmcculloug 
As you and I discussed, because auto_approval is handled differently in services, we don't need to  handle it here.